Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Build stage
+FROM python:3.10-slim AS build
```

Comment:
Consider adding an ARG for the Python base image version to allow flexibility and meet the task requirements.